### PR TITLE
Add `create-rule` script

### PR DESCRIPTION
**Pull request by fisker**

### CODE REVIEW

File: scripts/create-rule.mjs [hunk 1]
```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env node
+import fs from "fs";
```
Comment (sindresorhus) [inline scripts/create-rule.mjs:2]: Use single-quotes.

File: scripts/create-rule.mjs [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+		if (fs.existsSync(path.join(ROOT, file))) {
+			throw new Error(`"${file}" already exists.`);
+		}
+	}
+}
+
+function renderTemplate({ source, target, data }) {
+	const templateFile = path.join(DIRNAME, `template/${source}`);
+	const targetFile = path.join(ROOT, target);
+	const templateContent = fs.readFileSync(templateFile, "utf8")
```
Comment (sindresorhus) [inline scripts/create-rule.mjs:28]: Semicolons.

File: scripts/create-rule.mjs [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+	let ruleContent = `${RULE_INDENT}'unicorn/${id}': 'error',`;
+
+	const file = path.join(ROOT, "index.js");
+	const content = fs.readFileSync(file, "utf8");
+	const [before, rest] = content.split(RULE_START);
+	const [rules, after] = rest.split(RULE_END);
+
+	let lines = rules.split("\n");
+	if (!lines.every((line) => line.startsWith(RULE_INDENT))) {
+		throw 'Unexpected content in "index.js".';
```
Comment (sindresorhus) [inline scripts/create-rule.mjs:48]:
```suggestion
		throw 'Unexpected content in “index.js”.';
```

File: scripts/create-rule.mjs [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+		data
+	});
+	renderTemplate({
+		source: 'test.js.jst',
+		target: `test/${id}.js`,
+		data
+	});
+	updateIndex(id);
+	updateReadme(data);
+})().catch((error) => {
```
Comment (sindresorhus) [inline scripts/create-rule.mjs:211]:
```suggestion
})().catch(error => {
```

File: scripts/template/documentation.md.jst [hunk 5]
```diff
@@ -0,0 +1,19 @@
+# <%= description %>
+
+<!-- More detailed description -->
+<% if (fixable === true) { %>
```
Comment (sindresorhus) [inline scripts/template/documentation.md.jst:4]:
```suggestion
<% if (isFixable === 'true') { %>
```

File: scripts/template/documentation.md.jst [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+This rule is fixable.
+<% } else if (fixable === 'partly') { %>
+This rule is partly fixable.
+<% } %>
+## Fail
+
+```js
+const foo = "unicorn";
```
Comment (sindresorhus) [inline scripts/template/documentation.md.jst:12]:
```suggestion
const foo = 'unicorn';
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by sindresorhus** [comment]
Great idea 👍🏻

**Comment by fisker** [comment]
Why xo didn't catch style issues?

**Comment by sindresorhus** [comment]
It doesn't yet support `.mjs`.

File: scripts/template/documentation.md.jst (hunk 5 above)
Comment (fisker) [reply scripts/template/documentation.md.jst:4]: It's boolean, and it's named fixable in create-rule.js

**Comment by sindresorhus** [comment]
> It doesn't yet support .mjs.

XO is blocked on https://github.com/davidtheclark/cosmiconfig/issues/224

**Comment by fisker** [comment]
I don't understand isn't add `.mjs` [here](https://github.com/xojs/xo/blob/2346fef2fe8825bd937f9602268aa8977c1f94e2/lib/constants.js#L29) will make it check `.mjs` files?

**Comment by sindresorhus** [comment]
> I don't understand, isn't add mjs here will make it check .mjs files?

Yeah, I was mixing it up with the config format extension. I think that would do it, yes, but it needs testing and docs in XO.

File: scripts/template/documentation.md.jst [hunk 7]
```diff
@@ -0,0 +1,19 @@
+# <%= description %>
+
+<!-- More detailed description -->
```
Comment (sindresorhus) [inline scripts/template/documentation.md.jst:3]:
```suggestion
<!-- More detailed description. Remove this comment. -->
```

File: scripts/template/documentation.md.jst (hunk 5 above)
Comment (sindresorhus) [reply scripts/template/documentation.md.jst:4]: The quoting was a mistake. The point was the naming.

File: scripts/template/rule.js.jst [hunk 8]
```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+const MESSAGE_ID = '<%= id %>';
+const messages = {
+	[MESSAGE_ID]: 'Prefer "{{replacement}}" over "{{value}}".'
```
Comment (sindresorhus) [inline scripts/template/rule.js.jst:7]:
```suggestion
	[MESSAGE_ID]: 'Prefer `{{replacement}}` over `{{value}}`.'
```

File: scripts/template/rule.js.jst [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+					replacement: '🦄'
+				}<% if (fixable) { %>,
+				fix: fixer => fixer.replaceText(node, '\'🦄\'')<% } %>
+			});
+		}
+	}
+};
+
+const schema = [
+];
```
Comment (sindresorhus) [inline scripts/template/rule.js.jst:31]:
```suggestion
const schema = [];
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by fisker** [comment]
> I guess for now, you could just use the XO extensions option.

I've changed it to commonjs, I don't want change it again :smile:, I'll address the review point tomorrow.

**Pull request merged**